### PR TITLE
Synchronize the GCS bucket with RouteViews archives

### PR DESCRIPTION
**Pull request by yuhao-he**

### CODE REVIEW

File: pkg/utils/upload/upload.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	"cloud.google.com/go/storage"
+)
+
+// MD5FromGCS returns the MD5 encoding from the GCS object's metadata.
+func MD5FromGCS(ctx context.Context, hd *storage.BucketHandle, path string) (string, error) {
+	attrs, err := hd.Object(path).Attrs(ctx)
+	if err != nil {
+		return "", fmt.Errorf("failed to get attrs for obj: %v", err)
+	}
+	return hex.EncodeToString(attrs.MD5), nil
```
Comment (morrowc) [inline pkg/utils/upload/upload.go:21]: this does basically the same as fmt.Sprintf("%x", )
right? why not do the same thing (this hex.EncodeToString() or Sprintf) for this checksum return value? (in MD5FromHTTP)

File: pkg/synchronizer/synchronizer.go [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+			}
+			return nil
+		}, backoff.WithMaxRetries(backoff.NewConstantBackOff(s.retryInterval), s.retryCount))
+		if err != nil {
+			log.Error(err)
+			return
+		}
+
+		// Check if any archive is missing (two consecutive archives are over
+		// 30 minutes apart). Structured log for potential metric collection.
```
Comment (morrowc) [inline pkg/synchronizer/synchronizer.go:138]: you say 30 here, but check for 15 below (line 151)

**Review by morrowc — APPROVED** [review]

**Comment by morrowc** [comment]
ping on this

**Comment by morrowc** [comment]
(to make sure I didn't miss something I mean :) )

File: cmd/archive_synchronizer/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    ```shell
+    $   docker build -f cmd/archive_synchronizer/Dockerfile . -t us-docker.pkg.dev/public-routing-data-backup/cloudrun/archive-sync:latest
+    ```
+    -   Don't forget to upload it to the registries: 
+    ```shell
+    docker push
+        us-docker.pkg.dev/public-routing-data-backup/cloudrun/archive-sync:latest`
+    ```
+2.  Deploy the image by setting the output bucket `BIGQUERY_BUCKET` for
+    converted updates.
```
[the remark this answers is not in the document]
Comment (yuhao-he) [reply cmd/archive_synchronizer/README.md]: Done.

File: cmd/archive_synchronizer/sync.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+		HTTPRoot:        "http://routeviews.org",
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	now := time.Now()
+	start := now.Add(-*maxLapse)
+	end := now.Add(-*minLapse)
+	if *runHTTP {
```
[the remark this answers is not in the document]
Comment (yuhao-he) [reply cmd/archive_synchronizer/sync.go]: Done.

File: pkg/utils/upload/upload.go (hunk 1 above)
Comment (yuhao-he) [reply pkg/utils/upload/upload.go]: Oh, I copied this from the mass uploader - thought this is the right way to convert MD5.

File: pkg/synchronizer/synchronizer.go (hunk 2 above)
Comment (yuhao-he) [reply pkg/synchronizer/synchronizer.go]: Done.

**Pull request merged**